lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -973,20 +836,18 @@ DotOperandEncodingAttr::toLinearLayout(ArrayRef<int64_t> shape) const {
 std::optional<LinearLayout>
 toLinearLayout(ArrayRef<int64_t> shape, Attribute layout,
                std::optional<int32_t> elemBitWidth /*= std::nullopt*/) {
+  // Layouts are distributed or shared
   if (auto distributed = dyn_cast<DistributedEncodingTrait>(layout)) {
     return distributed.toLinearLayout(shape);
-  }
-  if (auto shared = dyn_cast<SharedEncodingAttr>(layout)) {
+  } else {
+    auto shared = cast<SharedEncodingAttr>(layout);
```

Comment:
Better to revert the change because third party backends may have other custom layouts